src/main/haskell/language-kore/src/Data/Kore/Substitution/Class.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE FlexibleContexts      #-}
```

Comment:
Could you add a test with a substitution `x -> something` and a pattern `and(exists x:x, x)`?